rmc/utils/generic.py

```diff
@@ -359,14 +376,24 @@ def get_ref_aa(
     )
 
     # Check to see if AA info is defined for all alleles associated with a locus/transcript
-    ht = ht.annotate(
-        all_aa_def=hl.all(hl.map(lambda x: hl.is_missing(x.ref_aa), ht.aa_info))
+    missing_aa_check_he_path = f"{TEMP_PATH_WITH_FAST_DEL}/missing_aa_check.he"
+    if overwrite_temp:
+        ht = ht.annotate(
+            all_aa_def=hl.all(hl.map(lambda x: hl.is_missing(x.ref_aa), ht.aa_info))
+        )
+        missing_aa_check = ht.aggregate(hl.agg.count_where(ht.all_aa_def))
+        missing_aa_check = hl.experimental.write_expression(
+            missing_aa_check, missing_aa_check_he_path, overwrite=True
+        )
+
+    missing_aa_check = hl.eval(
+        hl.experimental.read_expression(missing_aa_check_he_path)
     )
```

Comment:
wait, for the `~`:
```
        ht = ht.annotate(
            all_aa_def=hl.all(hl.map(lambda x: hl.is_missing(x.ref_aa), ht.aa_info))
        )
```
is first annotating a list of booleans based on whether the AA was missing (so `[False, False, False]` if all AA were defined) and then flattening it using `hl.all` to be a single boolean (so a single `False` in this case). the aggregate then counts any instances of True, which would indicate an AA was actually missing in the list, so not adding the `~` is the correct behavior, right? maybe the fix is actually renaming `all_aa_def` -> `any_aa_missing`?

Reply:
Oh yes, agree

Reply:
Wait. Don't we want `all_aa_def=hl.all(hl.map(lambda x: hl.is_defined(x.ref_aa), ht.aa_info))` and then `missing_aa_check = ht.aggregate(hl.agg.count_where(~ht.all_aa_def))`? Because `hl.all` is only True if all the elements are True. So any instances of `hl.all=True` will only arise if all AAs are missing in the list

Reply:
Lol yeah that works. So no `~`